src/action.js

```diff
@@ -1,68 +1,187 @@
 import Emitter from './emitter'
-import History from './history'
 import tag from './tag'
 
 import {
   inherit
 } from './utils'
 
-import {
-  STATES
-} from './constants'
+let uid = 0
 
 /**
- * Actions encapsulate the process of resolving an action creator. Create an
- * action using `Microcosm::push`:
+ * Actions represent the state of an action. As actions change, their
+ * associated history coordinates with Microcosm instances to update
+ * state. Create an Action using `Microcosm::push`
+ * @constructor
+ * @param {Function|string} command
+ * @param {string} [status] Starting status
  */
-export default function Action (command, status, history) {
+export default function Action (command, status) {
   Emitter.call(this)
 
-  this.history = history || new History()
-
-  this.id = this.history.getId()
+  this.id = uid++
   this.command = tag(command)
   this.timestamp = Date.now()
   this.children = []
 
   if (status) {
-    this.setStatus(status)
+    console.assert(this[status], 'Unexpected task status ' + status)
+    this[status]()
   }
 }
 
 inherit(Action, Emitter, {
-  status     : 'inactive',
-  payload    : undefined,
-  disabled   : false,
-  disposable : false,
-  parent     : null,
-  next       : null,
+  status   : 'inactive',
+  payload  : undefined,
+  disabled : false,
+  complete : false,
+  parent   : null,
+  next     : null,
+
+  /**
+   * Open state. The action has started, but has received no response.
+   * @param {*} [nextPayload]
+   */
+  open (nextPayload) {
+    this._setPayload.apply(this, arguments)
+    this._setStatus('open', false)
+
+    return this
+  },
+
+  /**
+   * Update state. The action has received an update, such as loading progress.
+   * @param {*} [nextPayload]
+   */
+  update (nextPayload) {
+    this._setPayload.apply(this, arguments)
+    this._setStatus('update', false)
+
+    return this
+  },
+
+  /**
+   * Resolved state. The action has completed successfully.
+   * @param {*} [nextPayload]
+   */
+  resolve (nextPayload) {
+    this._setPayload.apply(this, arguments)
+    this._setStatus('resolve', true)
+
+    return this
+  },
+
+  /**
+   * Failure state. The action did not complete successfully.
+   * @param {*} [nextPayload]
+   */
+  reject (nextPayload) {
+    this._setPayload.apply(this, arguments)
+    this._setStatus('reject', true)
+
+    return this
+  },
+
+  /**
+   * Cancelled state. The action was halted, like aborting an HTTP
+   * request.
+   * @param {*} [nextPayload]
+   */
+  cancel (nextPayload) {
+    this._setPayload.apply(this, arguments)
+    this._setStatus('cancel', true)
 
+    return this
+  },
+
+  /**
+   * Subscribe to when an action opens.
+   * @param {Function} callback
+   * @param {*} [scope]
+   */
+  onOpen (callback, scope) {
+    this._callOrSubscribeOnce('open', callback, scope)
+    return this
+  },
```

Comment:
ActionButton and ActionForm:
https://github.com/vigetlabs/microcosm/blob/master/src/addons/action-button.js#L25
https://github.com/vigetlabs/microcosm/blob/master/src/addons/action-form.js#L53